kinesis.go

```diff
@@ -188,3 +196,11 @@ func (k *kinesis) getErrCount() int64 {
 func getLock(sem chan bool) {
 	sem <- true
 }
+
+func (k *kinesis) decreaseRequestLimit() {
+	k.limit = k.limit >> 1
```

Comment:
Since the change did not fix the issue when it was deployed. We may get to see what happens when it reaches zero.

I was thinking along the lines of we may hit a loop that ends up with the instance burning CPU and consuming throughput API limits when it was trying to prevent that. First glance through just now I think I saw a sleep statement when a zero result set size is returned. So hopefully no big.